core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java

```diff
@@ -538,6 +538,7 @@ private void closeOperators(int lastOperatorIndex)
                             operatorContext.getDriverContext().getTaskId());
                 }
                 finally {
+                    workProcessorOperatorContext.metrics.set(operator.getMetrics());
```

Comment:
you should also set `connectorMetrics` (if its first operator in pipeline) (+ test)

BTW:  I think you can just modify `io.trino.operator.WorkProcessorPipelineSourceOperator#workProcessorOperatorStateMonitor` method to report metrics after:
```
if (state.getType() == FINISHED) {
..
}
```
if

Reply:
> you should also set `connectorMetrics` (if its first operator in pipeline) (+ test)

Thanks, added in 2c2aca64e9.

> I think you can just modify `io.trino.operator.WorkProcessorPipelineSourceOperator#workProcessorOperatorStateMonitor` method

I am not sure, since `closeOperators` is dropping the closed operators in the `finally` clause:
https://github.com/trinodb/trino/blob/43e583667e8fbcb23038f932ccd32f5adc7c5c3f/core/trino-main/src/main/java/io/trino/operator/WorkProcessorPipelineSourceOperator.java#L543
Please let me know if you meant to refactor `closeOperators` method as well.